interop-cats/shared/src/main/scala/zio/interop/cats.scala

```diff
@@ -228,7 +228,11 @@ private class CatsConcurrent[R] extends CatsMonadError[R, Throwable] with Concur
     }
 
   override final def race[A, B](fa: RIO[R, A], fb: RIO[R, B]): RIO[R, Either[A, B]] =
-    fa.map(Left(_)).interruptible raceFirst fb.map(Right(_)).interruptible
+    (fa.map(Left(_)).interruptible raceWith fb.map(Right(_)).interruptible)(
```

Comment:
I think you need to use `.run` so that a failure of the first effect will stop the second, as in the implementation of `raceFirst` in ZIO https://github.com/zio/zio/blob/5375a0cd83fc5736d008b613927c747ec8336df1/core/shared/src/main/scala/zio/ZIO.scala#L1388